solvers/gsroptim.py

```diff
@@ -0,0 +1,45 @@
+from benchopt import BaseSolver
+from benchopt import safe_import_context
+
+with safe_import_context() as import_ctx:
+    import warnings
+    from sklearn.exceptions import ConvergenceWarning
+    from scipy import sparse
+    from gsroptim.lasso import lasso_path
+
+
+class Solver(BaseSolver):
+    name =  "gsroptim"
+    stopping_strategy = "iteration"
+
+    install_cmd = "conda"
+    requirements = [
+        'pip:git+https://github.com/EugeneNdiaye/Gap_Safe_Rules.git@master'
+    ]
+    references = [
+        "E. Ndiaye, O. Fercoq, A. Gramfort and J. Salmon, JMLR, "
+        '"Gap Safe screening rules for sparsity enforcing penalties", '
+        "vol. 18, pp. 1-33 (2017)"
+    ]
+
+    def skip(self, X, y, lambdas, fit_intercept):
```

Comment:
are you sure that sparse data is not supported ?

Reply:
i've checked the docstring. It expects an array of shape `(n_samples, n_features)`. 
I don't think it is optimized for sparse matrices.